gradio/queueing.py

```diff
@@ -32,6 +37,9 @@ def __init__(
         self.token: str | None = None
         self.progress: Progress | None = None
         self.progress_pending: bool = False
+        with new_event_lock:
+            self.idx = Event.idx
+            Event.idx += 1
```

Comment:
This is still not thread-safe, it may be async-safe but not thread-safe.

I would consider a lockless system, e.g. `self.idx = uuid.uuid4()` or [an ULID](https://github.com/valohai/ulid2) if you need chronological ordering.

Reply:
Why isn't it thread safe? (I'm using a threading lock now.)
I did want the idx to be a counter so in the traffic stats sidepanel, every task shows which event number they are. I think this would be helpful for watching the queue status.